components/DateTimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,82 @@
+'use client';
+
+import { useState, useCallback } from 'react';
+
+import InputField from '@/components/InputField/InputField';
+
+/**
+ * TimePicker 컴포넌트는 시간을 선택할 수 있는 기능을 제공합니다.
+ * @param {string} width - optional, TimePicker 컴포넌트의 너비를 설정합니다. (기본 값은 w-full입니다.)
+ * @returns {JSX.Element} TimePicker 컴포넌트를 반환합니다.
+ */
+
+export default function TimePicker({ width = 'full' }: { width?: string }) {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [time, setTime] = useState<string>('');
+  const [amPm, setAmPm] = useState<'am' | 'pm'>('am');
+
+  const timeSlots = useCallback(() => {
+    const slots: string[] = [];
+    for (let i = 1; i <= 12; i++) {
+      slots.push(`${String(i)}:00`);
+      slots.push(`${String(i)}:30`);
+    }
+    return slots;
+  }, []);
+
+  return (
```

Comment:
여기는 이렇게 버튼이 공통 스타일을 사용하니, 스타일 객체 변수와 함수를 사용해서 좀 더 깔끔하게 만들어볼 수 있을 것 같아요~
그리고 hover와 transition-all은 취향껏 넣은거니 참고만 해주세요!


```suggestion
  const timeButtonStyles = {
    default:
      'h-pr-40 w-pr-78 rounded-xl bg-b-primary text-14m hover:bg-brand-primary hover:text-white transition-all',
    active: 'bg-brand-primary text-white',
    inactive: 'bg-b-secondary text-t-default',
  };

  const getButtonClass = (period: 'am' | 'pm') => {
    return classNames(timeButtonStyles.default, {
      [timeButtonStyles.active]: amPm === period,
      [timeButtonStyles.inactive]: amPm !== period,
    });
  };

```

Reply:
오.. 확실히 이 편이 더 코드가 깔끔해지네여